runners/flink/src/main/java/org/apache/beam/runners/flink/translation/wrappers/streaming/state/FlinkStateInternals.java

```diff
@@ -571,6 +573,27 @@ public void clear() {
         throw new RuntimeException("Error clearing state.", e);
       }
     }
+
+    @Override
+    public boolean equals(@Nullable Object o) {
+      if (this == o) {
+        return true;
+      }
+      if (o == null || getClass() != o.getClass()) {
+        return false;
+      }
+
+      final FlinkOrderedListState<?> that = (FlinkOrderedListState<?>) o;
+
+      return namespace.equals(that.namespace) && stateId.equals(that.stateId);
+    }
+
+    @Override
+    public int hashCode() {
+      int result = namespace.hashCode();
```

Comment:
Can use `Objects.hashCode`

Reply:
@kennknowles 
The current implementation follows the same hashCode semantics used in FlinkStateInternals. I'm a bit unclear whether suggesting `Objects.hashCode` means we should replace the current implementation, or if you're suggesting that overriding `hashCode` is unnecessary altogether.

Reply:
Overriding equality and hashcode should always occur together, so it is necessary to override `hashCode`.

I was just suggesting this pattern instead of doing your own math: https://github.com/apache/beam/blob/7c86bf3103fec1d3a6a3a3cbe6ace8a14a85b723/sdks/java/core/src/main/java/org/apache/beam/sdk/testing/SuccessOrFailure.java#L88